src/common/schedulers/slurm_commands.py

```diff
@@ -467,53 +667,34 @@ def _parse_nodes_info(slurm_node_info):
     for node in node_info:
         lines = node.strip().splitlines()
         if lines:
-            slurm_nodes.append(SlurmNode(*lines))
+            if is_static_node(lines[0]):
+                node = StaticNode(*lines)
+                slurm_nodes.append(node)
+            else:
+                node = DynamicNode(*lines)
+                slurm_nodes.append(node)
     return slurm_nodes
 
 
-def get_nodes_type(nodes):
-    """Get node type given nodename. Node type format: NodeType(instancetype, queue_name)."""
-    nodes_types = set()
-    for node in nodes:
-        queue_name, _, instance_name = parse_nodename(node.name)
-        nodes_types.add(NodeType(instance_name, queue_name))
-    return nodes_types
-
-
-def get_inactive_partition_names(partitions):
+def get_inactive_partition_names(partition_names_map):
```

Comment:
I would imagine we won't need this anymore? The partition object should have functions to check if it has nodes that are currently running jobs and if it's inactive, so when we try to find partitions to disable we can just do
```
for part, failures in self._partitions_protected_failure_count_map.items()
    if part.is_active and failures >= self._config.protected_failure_count and not part.has_running_job():
         partitions_to_disable.append(part)
```